src/kava-ads.js

```diff
@@ -0,0 +1,127 @@
+    this._eventManager.listen(this._player, this._player.Event.AD_CLICKED, () => this._onAdClicked());
+  }
+
+  _onAdLoaded(event: FakeEvent): void {
+    this._kava.logger.debug('_onAdLoaded', event);
+    this._model.updateModel({adId: event.payload.ad._id});
+    this._model.updateModel({adTitle: event.payload.ad._title});
+    this._model.updateModel({adPosition: event.payload.ad._position});
+    this._model.updateModel({adSystem: event.payload.ad._system});
+    this._model.updateModel({advertiserName: event.payload.ad._advertiserName});
+    this._model.updateModel({adBreakType: event.payload.adType});
+    this._model.updateModel({adImpressionTimeStamp: Date.now()});
+    this._sendAnalytics(KavaAdEventModel.AD_IMPRESSION);
+  }
+
+  _onAdBuffering(): void {
+    this._sendAnalytics(KavaAdEventModel.AD_BUFFER_START);
+    this._isBuffering = true;
+  }
+  _onAdClicked(): void {
+    this._sendAnalytics(KavaAdEventModel.AD_CLICKED);
+  }
+
+  _onAdCompleted(): void {
+    this._kava.logger.debug('_onAdCompleted');
+    this._sendAnalytics(KavaAdEventModel.AD_COMPLETED);
+    this._clearAdStartedModelData();
+    this._model.updateModel({adCurrentTime: 0});
+    this._model.updateModel({adCurrentTime: 0});
+  }
+
+  _clearAdStartedModelData() {
+    this._model.updateModel({adId: NaN});
+    this._model.updateModel({adTitle: ''});
+    this._model.updateModel({adPosition: NaN});
+    this._model.updateModel({adSystem: ''});
+    this._model.updateModel({advertiserName: ''});
+    this._model.updateModel({adImpressionTimeStamp: 0});
+    this._isBuffering = false;
+  }
+  _onAdSkipped(): void {
+    this._kava.logger.debug('_onAdSkipped');
+    this._sendAnalytics(KavaAdEventModel.AD_SKIPPED);
+    this._clearAdStartedModelData();
+    this._model.updateModel({adCurrentTime: 0});
+  }
+  _onAdStarted(): void {
+    this._kava.logger.debug('_onAdStarted');
+    this._sendAnalytics(KavaAdEventModel.AD_STARTED);
+  }
+
+  _onAdFirstQuartile(): void {
+    this._sendAnalytics(KavaAdEventModel.AD_FIRST_QUARTILE);
+  }
+
+  _onAdMidPoint(): void {
+    this._sendAnalytics(KavaAdEventModel.AD_MID_POINT);
+  }
+
+  _onAdThirdQuartile(): void {
+    this._sendAnalytics(KavaAdEventModel.AD_THIRD_QUARTILE);
+  }
+
+  _onAdBreakEnd(): void {
+    this._kava.logger.debug('_onAdBreakEnd');
+    this._model.updateModel({adBreakType: ''});
+  }
+
+  _onAdProgress(event: FakeEvent): void {
+    this._model.updateModel({adCurrentTime: event.payload.adProgress.currentTime});
+    if (this._isBuffering) {
+      this._sendAnalytics(KavaAdEventModel.AD_BUFFER_END);
+      this._isBuffering = false;
+    }
+  }
+
+  _onAdBreakStarted(event: FakeEvent): void {
+    this._kava.logger.debug('_onAdBreakStarted', event.payload.adBreak._type);
+    // the type might be deprecated in the impression so this will be a fallback
+    this._model.updateModel({adBreakType: event.payload.adBreak._type});
```

Comment:
if this is deprecated then don't use it, please remove and get it from the supported event